File: src/main/java/loxal/common/auth/AuthInfoSvc.java
/*
 * Copyright 2012 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.common.auth;

import com.google.gwt.user.client.rpc.RemoteService;
import com.google.gwt.user.client.rpc.RemoteServiceRelativePath;

@RemoteServiceRelativePath("AuthInfo")
public interface AuthInfoSvc extends RemoteService {
    /**
     * Google Authentication Service
     *
     * @param requestUri Request originator
     * @return Properties of the authenticated user
     */
    AuthInfo getAuthInfo(String requestUri);
}

File: src/main/kotlin/loxal/common/auth/AuthInfoSvc.kt
/*
 * Copyright 2015 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package loxal.common.auth

RemoteServiceRelativePath("AuthInfo")
public interface AuthInfoSvc : RemoteService {
    /**
     * Google Authentication Service
     *
     * @param requestUri Request originator
     * @return Properties of the authenticated user
     */
    public fun getAuthInfo(requestUri: String): AuthInfo
}